### PR TITLE
DOC: special: add reference for orthogonal polynomial functions

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

None

#### What does this implement/fix?

Add references to:

- The `eval_*` functions
- The `roots_*` functions

This is basically a massive "find the right equation in Abramowitz and Stegun for the formulas in the docstring".

#### Additional information

Also fix:

- a spacing issue in the `eval_laguerre` docstring
- a connection formula in the `eval_chebyc` docstring

These were uncovered while adding the references.

### CODE REVIEW

File: scipy/special/orthogonal.py [hunk 1]
```diff
@@ -1033,11 +1070,11 @@ def _newton(n, x_initial, maxit=5):
...
     r"""Gauss-Hermite (physicst's) quadrature for large n.
 
-    Computes the sample points and weights for Gauss-Hermite quadrature.
-    The sample points are the roots of the n-th degree Hermite polynomial,
+    Computes the sample points and weights for GaussHermite quadrature.
+    The sample points are the roots of the nth degree Hermite polynomial,
     :math:`H_n(x)`.  These sample points and weights correctly integrate
-    polynomials of degree :math:`2n - 1` or less over the interval
-    :math:`[-\infty, \infty]` with weight function :math:`f(x) = e^{-x^2}`.
+    polynomials of degree :math:`2n  1` or less over the interval
```
Comment (rgommers) [inline scipy/special/orthogonal.py]: what's going on with removing all `-`'s in this docstring?

**Comment by rgommers** [comment]
LGTM modulo my one comment. Of course I didn't double check all the equation numbers (also, I don't own a Abramowitz and Stegun copy) but I trust those are right.

File: scipy/special/orthogonal.py (hunk 1 above)
Comment (person142) [reply scipy/special/orthogonal.py]: Whoa, I have no idea what happened there! Let me fix that...

**Comment by rgommers** [comment]
Thanks @person142, merged straight away to not run CI twice in a row (it was green already)

**Pull request merged**